lib/utility/recipe_puppy_parser.rb

```diff
@@ -0,0 +1,74 @@
+module Utility
+  class RecipePuppyParser
+    BASE_URL = 'http://www.recipepuppy.com/api/'.freeze
+    ALL_RECIPES_URL = 'http://www.recipepuppy.com/api/?q=&p='.freeze
+    RECIPE_QUERY_BASE = 'http://www.recipepuppy.com/api/?q='.freeze
+    MAX_PAGES = 100.freeze
+
+    def query_for(recipe)
+      recipe.gsub!(/\s/, '+')
+      MAX_PAGES.times do |page|
```

Comment:
This is making me think...

When we eventually implement a search bar, do we want 1000 results? I don't really think it's very practical. We only want maybe 10-20. I think it might be a good idea to extract the number of times the call occurs into an argument so that our tasks can pass MAX_PAGES there instead, and when we write our search controller, it can just pass a 1 or a 2